controllers/disruption_controller.go

```diff
@@ -915,7 +927,10 @@ func (r *DisruptionReconciler) getEligibleTargets(instance *chaosv1beta1.Disrupt
 		}
 
 		// add target if eligible
-		eligibleTargets = append(eligibleTargets, target)
+		eligibleTargets[target] = chaosv1beta1.TargetInjection{
+			InjectionStatus: chaostypes.DisruptionInjectionStatusNotInjected,
+			Since:           time.Now().Unix(),
```

Comment:
Once again, we have a `time.Now()` call here but keep in mind that a disruption can be reconciled very often so it does not necessarily correlate with a change on the injection status.